### ISSUE TITLE
sum method by row or column

### ISSUE THREAD

**Issue by JeffersonH44** [mljs/matrix#47]
It would be nice to have a parameter at the sum method that allows to get the sum at each row or column, for example:
```javascript
// assuming that this is a Matrix instance
var a = [[3, 3], [2, 2]];
a.sum('rows'); // [[3], [2]]
a.sum('columns'); //[[5, 5]] 
```

**Comment by targos** [comment]
Do you know of a similar method in matlab or python ?

**Comment by JeffersonH44** [comment]
on numpy there is an argument 'axis' that makes the sum by the axis because they support multidimentional arrays, but for us I think is ok with the string because we only have a two dimentional arrays.

https://docs.scipy.org/doc/numpy/reference/generated/numpy.sum.html

**Comment by targos** [comment]
your example feels wrong:
`a.sum('rows'); // [[3], [2]]`
I would expect the result to be `[[6], [4]]`

**Comment by JeffersonH44** [comment]
sorry, yes that should be the result.

**Comment by targos** [comment]
I'm OK for extending the `sum` method as you suggest but should we return a 2D matrix or just a 1D array ?

**Comment by JeffersonH44** [comment]
it's better a 2D matrix because it's easier to chain with other operations

**Comment by targos** [comment]
OK. Can you open a PR?

**Comment by targos** [comment]
sum is here: https://github.com/mljs/matrix/blob/master/src/abstractMatrix.js#L890

**Comment by JeffersonH44** [comment]
Ok

**Comment by targos** [comment]
Available in v2.2.0
